SPECS/libkcapi/libkcapi.spec

```diff
@@ -0,0 +1,567 @@
+Utility applications that are provided with %{name}.  This includes
+tools to use message digests, symmetric ciphers and random number
+generators implemented in the Linux kernel from command line.
+
+%if %{with test_package}
+%package        tests
+Summary:        Testing scripts for the %{name} package
+Requires:       %{name}%{?_isa} = %{version}-%{release}
+Requires:       %{name}-tools%{?_isa} = %{version}-%{release}
+Requires:       coreutils
+Requires:       openssl
+Requires:       perl-interpreter
+%if %{with replace_hmaccalc}
+Requires:       %{name}-hmaccalc%{?_isa} = %{version}-%{release}
+%endif
+%if %{with replace_coreutils}
+Requires:       %{name}-checksum%{?_isa} = %{version}-%{release}
+%endif
+
+%description    tests
+Auxiliary scripts for testing %{name}.
+%endif
+
+
+%prep
+%autosetup -p 1 -S git
+
+%if %{with_sysctl_tweak}
+
+cat << EOF > %{sysctl_prio}-%{name}-optmem_max.conf
+# See the 'README.%{distroname_ext}' file shipped in %%doc
+# with the %{name} package.
+#
+# See '%{_sysctldir}/50-default.conf',
+# if you need or want to override this preset.
+
+# Increase the ancillary buffer size per socket.
+net.core.optmem_max = %{sysctl_optmem_max}
+EOF
+%endif
+
+%{_bindir}/autoreconf -fiv
+
+
+%build
+%configure               \
+  --libdir=/%{_lib}      \
+  --disable-silent-rules \
+  --enable-kcapi-encapp  \
+  --enable-kcapi-dgstapp \
+  --enable-kcapi-hasher  \
+  --enable-kcapi-rngapp  \
+  --enable-kcapi-speed   \
+  --enable-kcapi-test    \
+  --enable-shared        \
+  --enable-static        \
+  --enable-sum-prefix=   \
+  --enable-sum-dir=/%{_lib} \
+  --with-pkgconfigdir=%{_libdir}/pkgconfig
+
+
+%install
+%make_install
+
+# Install sysctl.d preset.
+mkdir -p %{buildroot}%{_sysctldir}
+install -Dpm 0644 -t %{buildroot}%{_sysctldir} \
+  %{sysctl_prio}-%{name}-optmem_max.conf
+
+
+# Install replacement tools, if enabled.
+%if !%{with replace_coreutils}
+rm -f                            \
+  %{buildroot}%{_bindir}/md5sum       \
+  %{buildroot}%{_bindir}/sha*sum
+%endif
+
+%if !%{with replace_fipscheck}
+rm -f %{buildroot}%{_bindir}/fips*
+%endif
+
+%if !%{with replace_hmaccalc}
+rm -f %{buildroot}%{_bindir}/sha*hmac
+%endif
+
+# We don't ship autocrap dumplings.
+find %{buildroot} -type f -name "*.la" -delete -print
+
+# HMAC checksums are generated during __spec_install_post.
+%{_bindir}/find %{buildroot} -type f -name '*.hmac' -print -delete
+
+# Remove 0-size files.
+%{_bindir}/find %{buildroot} -type f -size 0 -print -delete
+
+
+%check
+
+# On some arches `/proc/sys/net/core/optmem_max` is lower than 20480,
+# which is the lowest limit needed to run the testsuite.  If that limit
+# is not met, we do not run it.
+%if %{test_optmem_max} >= 20480
+# Skip the testsuite on old kernels.
+%if %{lua:print(rpm.vercmp(posix.uname('%r'), '5.1'));} >= 0
+# Real testsuite.
+pushd test
+ENABLE_FUZZ_TEST=1 \
+NO_32BIT_TEST=1    \
+  ./test-invocation.sh
+popd
+%endif
+%endif
+
+
+%ldconfig_scriptlets
```

Comment:
As far as I can remember, the `%ldconfig*` macros don't work on `1.0-dev` and have to be replaced with this pair:
```
%post -p /sbin/ldconfig
%postun -p /sbin/ldconfig
```